src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -19,6 +19,7 @@ internal static partial class TarHelpers
     {
         internal const short RecordSize = 512;
         internal const int MaxBufferLength = 4096;
+        internal const long MaxSizeLength = (1L << 33) - 1;
```

Comment:
```suggestion
        internal const long MaxSizeLength = (1L << 33) - 1; // Max value of 11 octal digits = 2^33 - 1 or 8 Gb.
```

Reply:
The largest possible number you can fit in 11 octal digits is `77_777_777_777`, which is equivalent to `8,589,934,591` in decimal (which can be obtained with the above shift and subtraction operations).

Also, the size field is 12 bytes in length, but one char is reserved for a null char. So we only have 11 chars to store an actual octal number.